src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -107,29 +115,49 @@ private void updateSection(Section existingSection, Section newSection, boolean
 
     private void addSectionWithConnectedUpStation(Section newSection) {
         Station newUpStation = newSection.getUpStation();
-
-        for (Section section : sections) {
-            if (section.getUpStation().equals(newUpStation)) {
-                updateSection(section, newSection, true);
-                return;
-            }
-        }
-
-        // 끝에 추가되는 경우
-        sections.add(newSection);
+        sections.stream()
+                .filter((section) -> section.getUpStation().equals(newUpStation))
+                .findFirst()
+                .ifPresentOrElse(
+                        (section) -> updateSection(section, newSection, true),
+                        () -> sections.add(newSection)
+                );
     }
 
     private void addSectionWithConnectedDownStation(Section newSection) {
         Station newDownStation = newSection.getDownStation();
+        sections.stream()
+                .filter((section) -> section.getDownStation().equals(newDownStation))
+                .findFirst()
+                .ifPresentOrElse(
+                        (section) -> updateSection(section, newSection, false),
+                        () -> sections.add(0, newSection)
+                );
+    }
 
-        for (Section section : sections) {
-            if (section.getDownStation().equals(newDownStation)) {
-                updateSection(section, newSection, false);
-                return;
-            }
-        }
+    private void mergeSections(Section previousSection, Section nextSection) {
+        Section mergedSection = Section.createSection(
+                previousSection.getLine(),
+                previousSection.getUpStation(),
+                nextSection.getDownStation(),
+                previousSection.getSectionDistance().plus(nextSection.getSectionDistance())
+        );
```

Comment:
두 지하철 구간을 합치는 정적 팩토리 메서드를 만들어봐도 좋을 것 같아요.
```suggestion
        Section mergedSection = Section.mergeSection(previousSection, nextSection);
```

Reply:
두 구간을 합치는 행위는 `sections = new ArrayList<>();`내에서 `add`, `remove` 를 처리하는 것이고, 구간 생성 자체는 새로운 구간이든 병합되는 구간이든 동일하게 생성이 되는 것 같은데요. 결국 `Section.mergeSection(previousSection, nextSection);`이 Section을 생성한다는 것은 구간을 생성만 하고, `add`, `remove`를 별도로 처리한다는 의미 같아요. 그래서 꼭 팩토리 메서드가 필요한 부분 같지는 않은데, 제가 놓치고 있는 부분이 있을까요?

Reply:
아 반드시 꼭 추가하실 필요는 없구요 기존에 일반적으로 새로운 구간 하나를 생성하는 `createSection()`과, 두 구간을 합쳐서 하나의 구간으로 생성하는 걸 이름으로 구분해보면 어떨까 해서 제안 드려봤습니다 😄